Yosemite/Yosemite/PointOfSale/PointOfSaleCouponService.swift

```diff
@@ -68,6 +69,20 @@ public final class PointOfSaleCouponService: PointOfSaleCouponServiceProtocol {
             return .init(items: refreshedCoupons, hasMorePages: false)
         }
     }
+
+    @MainActor
+    public func enableCoupons() async {
```

Comment:
Should it throw an error so the controller could set some sort of `enableCouponErrorState` with an option to retry enabling it?

```suggestion
    public func enableCoupons() async throws {
```

Reply:
Yes, this is tracked by WOOMOB-255 but we can already change the signature of this to be async throws: https://github.com/woocommerce/woocommerce-ios/pull/15478/commits/fb126bc37fc221c755e838f16faf849f0a07b5a0

Reply:
WOOMOB-255 doesn't mention error handling. If it's going to be handled there we should update the task.

Reply:
Oh, you're totally right. I've logged this in a new WOOMOB-267. We might be able to close both issues in the same PR, but I'll keep the tasks separate for now as are more approachable.